src/modules/editSignup/actions.js

```diff
@@ -41,6 +41,7 @@ export const getSignupAndEventAsync = (id, editToken) => (dispatch) => {
   return request('GET', `${PREFIX_URL}/api/signups/${id}?editToken=${editToken}`)
     .then(res => JSON.parse(res.body))
     .then((res) => {
+      if (res.signup === null) throw new Error('signup not found');
```

Comment:
Combined with the error handling changes in `getSignupAndEvent.js` we can intentionally throw here to make the `<EditSignup />` display a proper message to the user